Establish if Fuzzy is cross-platform

I'm not sure if the tests are currently broken or just on my operating system

```
self = <test_signature_generator.TestPSig testMethod=test_generate_signatures>

    def test_generate_signatures(self):
        """Test a multi-stragegy signatures."""
        dtuple = ('Joyce', 'Wang', 2134)
        signatures = [
            [
                {'type': 'feature-value', 'feature_idx': 0},
                {'type': 'feature-value', 'feature_idx': 1},
            ],
            [
                {'type': 'soundex', 'feature_idx': 0},
                {'type': 'soundex', 'feature_idx': 1},
            ]
        ]
        signatures = generate_signatures(signatures, dtuple)
>       assert signatures == {"J2W52", "JoyceWang"}
E       AssertionError: assert {'', 'JoyceWang'} == {'J2W52', 'JoyceWang'}
E         Extra items in the left set:
E         ''
E         Extra items in the right set:
E         'J2W52'
E         Use -v to get the full diff
```

If the tests are working for you we may need to check Fuzzy's support for Windows/MacOS/Linux

Same version for me (currently on "linux" (Windows subsystem for Linux). Another test failure has a different clue:

```
________________________________________________ TestPSig.test_soundex _________________________________________________
self = <test_signature_generator.TestPSig testMethod=test_soundex>

    def test_soundex(self):
        """Test signatures generated by soundex."""
        dtuple = ('Joyce', 'Wang', 2134)
        signature_strategies = [[{'type': 'soundex', 'feature_idx': 0},
                                {'type': 'soundex', 'feature_idx': 1},]]
>       signatures = generate_signatures(signature_strategies, dtuple)

tests/test_signature_generator.py:31:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
blocklib/signature_generator.py:141: in generate_signatures
    s = func(**config)
blocklib/signature_generator.py:79: in generate_by_soundex
    return soundex(feature)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   UnicodeDecodeError: 'ascii' codec can't decode byte 0xbb in position 1: ordinal not in range(128)

src/fuzzy.pyx:230: UnicodeDecodeError
```

--- a/blocklib/signature_generator.py
+++ b/blocklib/signature_generator.py
@@ -1,6 +1,6 @@
 from typing import Any, Callable, Dict, List, Sequence
 
-import fuzzy
+from metaphone import doublemetaphone
 
 
 def generate_by_feature_value(attr_ind: int, dtuple: Sequence):
@@ -63,9 +63,8 @@ def generate_by_metaphone(attr_ind: int, dtuple: Sequence):
 
     """
     feature = dtuple[attr_ind]
-    metaphone = fuzzy.DMetaphone()
-    phonetic_encoding = metaphone(feature)
-    return ''.join(p.decode() for p in phonetic_encoding if p is not None)
+    phonetic_encoding = doublemetaphone(feature)
+    return ''.join(phonetic_encoding)
 
 
 #################################################

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,7 @@
 bitarray==1.2.1
-Fuzzy==1.2.2
 jsonschema==3.2.0
 numpy==1.18.1
+metaphone==0.6
 tqdm==4.43.0
 pytest==5.3.5
 pytest-cov==2.8.1

--- a/setup.py
+++ b/setup.py
@@ -5,8 +5,8 @@ with open('README.md', 'r', encoding='utf-8') as f:
 
 requirements = [
     "jsonschema>=3.1.1",
-    "Fuzzy>=1.2.2",
     "numpy>=1.17.2",
+    "metaphone>=0.6",
     "tqdm>=4.36.1",
     "bitarray>=1.2.0"
 ]

--- a/tests/test_signature_generator.py
+++ b/tests/test_signature_generator.py
@@ -58,7 +58,7 @@ class TestPSig:
         'SM0XMT'
 
     def test_generate_signatures(self):
-        """Test a multi-stragegy signatures."""
+        """Test a multi-strategy signatures."""
         dtuple = ('Joyce', 'Wang', 2134)
         signatures = [
             [
@@ -71,7 +71,7 @@ class TestPSig:
             ]
         ]
         signatures = generate_signatures(signatures, dtuple)
-        assert signatures == {"JKAKANKFNK", "JoyceWang"}
+        assert signatures == {"JSASANKFNK", "JoyceWang"}
 
     def test_invalid_signature_type(self):
         dtuple = ('Joyce', 'Wang', 2134)

